Application/Client/src/app/card-list/components/card-list.component.ts

```diff
@@ -16,22 +20,84 @@ let tpls = require('../tpls/card-list.component.html').toString();
 export class CardListComponent{
 
     public showProductInfoPage: boolean = false;
-    constructor(public appState: AppState, private _settingsService : SettingsService, private _router:Router) {}
+    private wishListPostUrl:string = '';
+    private wishListCondtionalUrl:string = '';
+    private emailId:string = '';
+    private filterCategoryUrl:string = '';
+    private GetUserWishList: string = '';
+    private DeleteUserWishListUrl: string = '';
+    private isLoading:boolean = false;
 
     @Input() cards;
 
-    private isLoading:boolean = false;
-    
+    constructor(
+        public appState: AppState,
+        private _settingsService: SettingsService,
+        private _router: Router,
+        private _cservice: CService,
+        private _cookieService: CookieService
+    ) {
+        this.wishListPostUrl = _settingsService.getPath('wishListPostUrl');
+        this.filterCategoryUrl = _settingsService.getPath('filterCategoryUrl');
+        this.GetUserWishList = _settingsService.getPath('GetUserWishList');
+        this.DeleteUserWishListUrl = _settingsService.getPath('DeleteUserWishListUrl');
+    }
+
+    ngOnChanges(changes:SimpleChanges){
+        if(changes['cards']){
```

Comment:
Compare new changes and old changes if they are different then only make api call